text/0011-fluent-in-prefs-design-review.md

```diff
@@ -0,0 +1,61 @@
+
+The engineering team met with a set of architects and domain experts (the "review team") to discuss risks and mitigations, which constituted a lightweight "design review" of the project's architecture. This document describes the concerns that the review team raised and the risk mitigations that the teams discussed.
+
+## Why Preferences
+
+The team chose preferences because they're text-heavy, include users' names (which exercises Fluent's support for grammatical gender), and because the precision of text matters in preferences to ensure users clearly understand the implications of changes. So preferences are a good way to prove the change to Fluent before adopting it more widely.
+
+Fluent is a big change to the localization process, and the team plans to validate the change by translating just four strings initially (the "name" strings in brand.dtd/properties?) and then verifying those changes across the localization toolchain.
+
+## Chrome Privileges
+
+The review team asked whether there are parts of preferences that don't have chrome privileges, and if that influences the use of Fluent. It seems like the only such part is the Firefox Accounts image (???), which loads over a network and is localized independently of Firefox, so it won't be an issue.
+
+## Search Incompatibility
+
+The search feature of about:preferences is an issue, as the way that it searches preference labels and descriptions is not compatible with the way that Fluent injects text into the page.  The teams discussed options and concluded that there is a straightforward way to update the search implementation to be compatible with Fluent.
+
+Specifically, FTL messages support arbitrary attributes, so an FTL message for a searchable preference control could include a "searchterms" attribute containing the text to search, i.e.
+
+```
+key1
+  .label = "click"
+  .searchterms = " … "
+```
+
+Attribute values can include references to other strings within the FTL file, so localizers could avoid string duplication if desired.  This model would also enable localizers to customize search terms to include extra keywords that aren't included in other strings related to the control.  Finally, this model supports English fallback for searches via the creation of a secondary *document.l10n* object with the `en` locale.
+
+The review team raised some concerns about the performance of runtime searches and suggested that the engineering team consider generating an index from search terms to the controls they match, either at build time or on first search.  Some participants suggested being careful to avoid prematurely optimizing, however. In the end, the teams didn't reach consensus on the optimal solution, and further research/experimentation is warranted here.
```

Comment:
Worth mentioning that we mostly just agreed that all problems here are solvable, so we didn't feel the need to come to consensus.